apps/accounts/tests/test_views.py

```diff
@@ -242,12 +256,34 @@ def patch(self, thing):
         self.addCleanup(patcher.stop)
         return patcher.start()
 
+    def initialize_session(self, session_data):
+        # This is taken from django's login method.
+        # https://github.com/django/django/blob/9d915ac1be1e7b8cfea3c92f707a4aeff4e62583/django/test/client.py#L541
+        engine = import_module(settings.SESSION_ENGINE)
+        request = HttpRequest()
+        request.session = engine.SessionStore()
+        request.session.update(session_data)
+        # Save the session values.
+        request.session.save()
+        # Set the cookie to represent the session.
+        session_cookie = settings.SESSION_COOKIE_NAME
+        self.client.cookies[session_cookie] = request.session.session_key
+        cookie_data = {
+            'max-age': None,
+            'path': '/',
+            'domain': settings.SESSION_COOKIE_DOMAIN,
+            'secure': settings.SESSION_COOKIE_SECURE or None,
+            'expires': None,
+        }
+        self.client.cookies[session_cookie].update(cookie_data)
```

Comment:
Does this work instead?
````
session = self.client.session  # activate Django's weird getter
session['thing'] = 'value'
session.save()

# This should be all you need to load the signed cookie session.
# Note that `session_key` is actually the session contents.
self.client.cookies[settings.SESSION_COOKIE_NAME] = session.session_key
````
I used this on webpay and it seemed to work. We used encrypted cookie sessions but that was just an extension of signed cookies.